app/controllers/home_controller.rb

```diff
@@ -0,0 +1,31 @@
+require 'async/websocket/adapters/rails'
+
+class HomeController < ApplicationController
+  def index
+    self.response = Async::WebSocket::Adapters::Rails.open(request) do |connection|
+      loop do
+        channel = connection.read.to_str.chomp
+
+        subscribe(channel) do |context|
+          Console.info(connection, "Subscribed", channel: channel)
+
+          loop do
+            event = context.listen
+            Protocol::WebSocket::TextMessage.generate(event).send(connection)
+            connection.flush
+          end
+        end
+      end
+    end
+  end
```

Comment:
This seems like domain logic, so I'm not sure if it's appropriate to write it in Controller. However, since it calls methods of `connection` instance, I still think Controller is the most suitable place for it, rather than Model or ServiceObject.

Reply:
I think the design is okay, as the controller is focused on this particular use case. I think if you ended up with a more complex design, or wanted to isolate the logic for the sake of testing, you might prefer to use a service object.